apps/aevm/lib/aevm_util.ex

```diff
@@ -0,0 +1,374 @@
+
+  @spec default_opts :: map()
+  def default_opts do
+    %{
+      :execute_calls => true
+    }
+  end
+
+  @spec stop_exec(map()) :: map()
+  def stop_exec(state) do
+    code = State.code(state)
+    State.set_pc(byte_size(code), state)
+  end
+
+  def sdiv(_value1, 0), do: 0
+  def sdiv(0, -1), do: AevmConst.neg2to255()
+
+  @spec sdiv(integer(), integer()) :: non_neg_integer()
+  def sdiv(value1, value2) do
+    <<svalue1::integer-signed-256>> = <<value1::integer-unsigned-256>>
+    <<svalue2::integer-signed-256>> = <<value2::integer-unsigned-256>>
+    div(svalue1, svalue2) &&& AevmConst.mask256()
+  end
+
+  @spec smod(integer(), integer()) :: non_neg_integer()
+  def smod(value1, value2) do
+    <<svalue1::integer-signed-256>> = <<value1::integer-unsigned-256>>
+    <<svalue2::integer-signed-256>> = <<value2::integer-unsigned-256>>
+    result = rem(rem(svalue1, svalue2 + svalue2), svalue2)
+    result &&& AevmConst.mask256()
+  end
+
+  def pow(op1, op2) when is_integer(op1) and is_integer(op2) and op2 >= 0, do: pow(1, op1, op2)
+
+  def pow(n, _, 0), do: n
+  def pow(n, op1, 1), do: op1 * n
+
+  @spec pow(integer(), integer(), integer()) :: integer()
+  def pow(n, op1, op2) do
+    square = op1 * op1 &&& AevmConst.mask256()
+    exp = op2 >>> 1
+
+    case op2 &&& 1 do
+      0 -> pow(n, square, exp)
+      _ -> pow(op1 * n, square, exp)
+    end
+  end
+
+  @spec exp(integer(), integer()) :: integer()
+  def exp(op1, op2) do
+    pow(op1, op2) &&& AevmConst.mask256()
+  end
+
+  @spec signed(integer()) :: integer()
+  def signed(value) do
+    <<svalue::integer-signed-256>> = <<value::integer-unsigned-256>>
+    svalue
+  end
+
+  @spec byte(integer(), integer()) :: integer()
+  def byte(byte, value) when byte < 32 do
+    byte_pos = 256 - 8 * (byte + 1)
+    mask = 255
+    value >>> byte_pos &&& mask
+  end
+
+  def byte(_, _), do: 0
+
+  @spec get_op_code(map()) :: integer()
+  def get_op_code(state) do
+    pc = State.pc(state)
+    code = State.code(state)
+    prev_bits = pc * 8
+
+    <<_::size(prev_bits), op_code::size(8), _::binary>> = code
+
+    op_code
+  end
+
+  @spec move_pc_n_bytes(integer(), map()) :: {integer(), map()}
+  def move_pc_n_bytes(bytes, state) do
+    old_pc = State.pc(state)
+    code = State.code(state)
+
+    curr_pc = old_pc + 1
+    prev_bits = curr_pc * 8
+    value_size_bits = bytes * 8
+    code_byte_size = byte_size(code)
+
+    value =
+      cond do
+        curr_pc > code_byte_size ->
+          0
+
+        curr_pc + bytes >= code_byte_size ->
+          extend = (curr_pc + bytes - code_byte_size) * 8
+          <<_::size(prev_bits), value::size(value_size_bits)>> = <<code::binary, 0::size(extend)>>
+          value
+
+        true ->
+          <<_::size(prev_bits), value::size(value_size_bits), _::binary>> = code
+          value
+      end
+
+    state1 = State.set_pc(old_pc + bytes, state)
+
+    {value, state1}
+  end
+
+  def create_account(_value, _area, state) do
+    # TODO
```

Comment:
Is this a forgotten `TODO`?

Reply:
No, it will be implemented in an iteration to come